### PR TITLE
[BD-18] switch from nose to pytest

### PR DESCRIPTION
**Pull request by emZubair**
## [BD-18] switch from nose to pytest

updated Makefile commands as per pytest

**Packates Removed**

- nose
- nose-ignore-docstring

**Packages Added**

- pytest-cov

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @emzubair! I've created [BLENDED-546](https://openedx.atlassian.net/browse/BLENDED-546) to keep track of it in Jira.   More details are on the [BD-18](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1593934815/BD-18+Testing+Consistency+Improvements) project page.


When this pull request is ready, tag your edX technical lead.

File: requirements/pip-tools.txt [hunk 1]
```diff
@@ -4,6 +4,9 @@
...
 #    make upgrade
 #
-click==7.0                # via pip-tools
-pip-tools==4.2.0
+click==7.1.2              # via pip-tools
+pip-tools==5.3.1
 six==1.10.0
+
+# The following packages are considered to be unsafe in a requirements file:
+# pip==20.2.2               # via pip-tools
```
Comment (pwnage101) [inline requirements/pip-tools.txt]: I feel like I've seen this before, but I can't remember if it's okay.  Is this normal?

**Review by pwnage101 — APPROVED** [review]
LGTM, just one question.

**Comment by emZubair** [comment]
`diff-cover` requires a branch name so I've specified `origin/master`, let me know if you want me to update it.

File: requirements/pip-tools.txt (hunk 1 above)
Comment (emZubair) [reply requirements/pip-tools.txt]: @pwnage101 those lines were added as a result of `make upgrade` command, updated.

**Review by pwnage101 — COMMENTED** [review]
Last I recall, this is still pending acceptance testing.  @estute were you able to get acceptance tests back up and running last week?

**Comment by emZubair** [comment]
@feanil Can you please review this `PR`?

**Review by estute — COMMENTED** [review]
Looks good to me. Sorry for the delay. I am running the acceptance tests for this branch now. I'll let you know when they have finished

**Comment by ziafazal** [comment]
@estute Thanks for reviewing it. Could you please merge it if acceptance tests passed?

**Comment by feanil** [comment]
@ziafazal I spoke with @estute he ran into some issues with testing that he's still looking into.  Hopefully they'll be resolved soon but just wanted to let you know he's still working on it.

**Comment by feanil** [comment]
@estute any update on this yet?

**Comment by ashultz0** [comment]
This PR dropped through the cracks a long time ago before analytics pipeline came to its current team so I am closing it. If you still have interest please rebase and reopen or just reach out to me. To reopen we will have to figure out review - edX does not currently run the pipeline so whatever is merged is probably of more interest to community members like OpenCraft.

**Comment by openedx-webhooks** [comment]
@emZubair Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-analytics-pipeline&entry.867055334=https://github.com/openedx/edx-analytics-pipeline/pull/828&entry.1484655318=https://github.com/emZubair&entry.752974735=2020-08-17+14:21&entry.1917517419=2022-07-07+18:24&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**